sw/Makefile

```diff
@@ -1,7 +1,7 @@
 TOOLCHAIN_PREFIX=riscv64-unknown-elf-
 
 %.elf: %.S link.ld
-	$(TOOLCHAIN_PREFIX)gcc -nostartfiles -march=rv32i -mabi=ilp32 -Tlink.ld -o$@ $<
+	$(TOOLCHAIN_PREFIX)gcc -nostartfiles -nostdlib -march=rv32i -mabi=ilp32 -Tlink.ld -o$@ $<
```

Comment:
Without `-nostdlib`, the build failed on Ubuntu 20.04:

```
dave@ubuntu:~/Shared/verilog/serv/fusesoc_libraries/serv/sw$ uname -a
Linux ubuntu 5.8.0-59-generic #66~20.04.1-Ubuntu SMP Thu Jun 17 11:14:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

dave@ubuntu:~/Shared/verilog/serv/fusesoc_libraries/serv/sw$ make blinky.hex
riscv64-unknown-elf-gcc -nostartfiles -march=rv32i -mabi=ilp32 -Tlink.ld -oblinky.elf blinky.S
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lc
/usr/lib/riscv64-unknown-elf/bin/ld: cannot find -lgloss
collect2: error: ld returned 1 exit status
make: *** [Makefile:4: blinky.elf] Error 1
```

Reply:
Got it. Surprised that it worked so far. I'm slowly beginning to realize that those gcc flags are more like hints and every gcc installation seems to have different defaults :)

Reply:
One more data point. When running on my Mac, it does *not* need `-nostdlib`. However, it is a different (newer) version of the compiler:

```
riscv64-unknown-elf-gcc (SiFive GCC-Metal 10.2.0-2020.12.8) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

The Ubuntu version (which I installed via `sudo apt install gcc-riscv64-unknown-elf`) is:

```
dave@ubuntu:~$ riscv64-unknown-elf-gcc --version
riscv64-unknown-elf-gcc () 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

In any case, `-nostdlib` *also* worked with the newer version on my Mac, so its seems like a good change.